runtime/interpreter/interpreter.go

```diff
@@ -3752,23 +3836,26 @@ func (interpreter *Interpreter) capabilityCheckFunction(
 			// which reads the stored value
 			// and performs a dynamic type check
 
-			if reference.ReferencedValue(interpreter) == nil {
-				return FalseValue
-			}
-
-			return TrueValue
+			return AsBoolValue(
+				reference.ReferencedValue(interpreter) != nil,
+			)
 		},
 		sema.CapabilityTypeCheckFunctionType(borrowType),
 	)
 }
 
-func (interpreter *Interpreter) GetCapabilityFinalTargetPath(
+type CapabilityTarget struct {
+	Path    PathValue
+	Address common.Address
+}
```

Comment:
Maybe add a note saying only one of these two fields can present at a given time

Reply:
Replaced this with an interface in ff01114bdbf86e0f7e876bcac45d63c4ff11836c to express that this is either one case